context-monitoring/src/main/java/de/atb/context/monitoring/monitors/messagebroker/MessageBrokerMonitor.java

```diff
@@ -0,0 +1,244 @@
+package de.atb.context.monitoring.monitors.messagebroker;
+
+/*-
+ * #%L
+ * ATB Context Monitoring Core Services
+ * %%
+ * Copyright (C) 2015 - 2021 ATB – Institut für angewandte Systemtechnik Bremen GmbH
+ * %%
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * #L%
+ */
+
+import de.atb.context.monitoring.analyser.messagebroker.MessageBrokerAnalyser;
+import de.atb.context.monitoring.config.models.DataSource;
+import de.atb.context.monitoring.config.models.DataSourceType;
+import de.atb.context.monitoring.config.models.Interpreter;
+import de.atb.context.monitoring.config.models.Monitor;
+import de.atb.context.monitoring.config.models.datasources.MessageBrokerDataSource;
+import de.atb.context.monitoring.events.MonitoringProgressListener;
+import de.atb.context.monitoring.index.Indexer;
+import de.atb.context.monitoring.models.IMessageBroker;
+import de.atb.context.monitoring.models.IMonitoringDataModel;
+import de.atb.context.monitoring.monitors.ThreadedMonitor;
+import de.atb.context.monitoring.parser.IndexedFields;
+import de.atb.context.monitoring.parser.messagebroker.IndexedMessageBrokerFields;
+import de.atb.context.monitoring.parser.messagebroker.MessageBrokerParser;
+import de.atb.context.services.faults.ContextFault;
+import de.atb.context.tools.ontology.AmIMonitoringConfiguration;
+import org.apache.lucene.document.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * WebServiceMonitor
+ *
+ * @author scholze
+ * @version $LastChangedRevision: 143 $
+ */
+public class MessageBrokerMonitor extends ThreadedMonitor<IMessageBroker, IMonitoringDataModel<?, ?>> implements MonitoringProgressListener<IMessageBroker, IMonitoringDataModel<?, ?>>, Runnable {
```

Comment:
not sure how this whole class will fit with e.g. rabbitmq, where you establish a connection to the broker and then basically just register a callback method to handle incoming messages.

See: https://github.com/rabbitmq/rabbitmq-tutorials/blob/master/java/Recv.java
and for a spring-based example: https://github.com/rabbitmq/rabbitmq-tutorials/blob/master/spring-amqp/src/main/java/org/springframework/amqp/tutorials/tut3/Tut3Receiver.java

Reply:
this is now adjusted so that it should be in line with suggested callback approach